ui/src/components/schemas/SchemaDetails.tsx

```diff
@@ -0,0 +1,342 @@
+import { Button, Card, Dropdown, Row, Space, Typography, message } from "antd";
+import { useCallback, useEffect, useState } from "react";
+import { generatePath, useNavigate, useParams } from "react-router-dom";
+import SyntaxHighlighter from "react-syntax-highlighter";
+import { a11yLight } from "react-syntax-highlighter/dist/esm/styles/hljs";
+
+import { z } from "zod";
+import { ErrorResult } from "../shared/ErrorResult";
+import { LoadingResult } from "../shared/LoadingResult";
+import { SiderLayoutContent } from "../shared/SiderLayoutContent";
+import { Schema as ApiSchema, getSchema } from "src/adapters/api/schemas";
+import { ReactComponent as ChevronDownIcon } from "src/assets/icons/chevron-down.svg";
+import { ReactComponent as CreditCardIcon } from "src/assets/icons/credit-card-plus.svg";
+import { Detail } from "src/components/schemas/Detail";
+import { SchemaTree } from "src/components/schemas/SchemaTree";
+import { useEnvContext } from "src/contexts/env";
+import { Json, JsonLdType, Schema } from "src/domain";
+import { ROUTES } from "src/routes";
+import {
+  APIError,
+  downloadJsonFromUrl,
+  getJsonLdTypesFromUrl,
+  getSchemaFromUrl,
+  processZodError,
+} from "src/utils/adapters";
+import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
+import { CARD_WIDTH } from "src/utils/constants";
+import { formatDate } from "src/utils/forms";
+import { AsyncTask, hasAsyncTaskFailed, isAsyncTaskStarting } from "src/utils/types";
+
+type JsonView = "formatted" | "jsonLdContext" | "jsonSchema";
```

Comment:
This repetition seems excessive - most of this is also present in `ImportSchemaPreview`. Please have a look at [this proposal](https://github.com/0xPolygonID/sh-id-platform/pull/262) to abstract and reuse these common components. We can probably refine this further.

Reply:
I consciously resisted the temptation of doing this abstraction. Most of the time it's ok to repeat some things initially, especially when we still don't know if these abstractions will make sense in the future. I will take a look at the PR but I also would like to hear @adrifdez and @elias-garcia's opinions on the proposal.

Reply:
I would say abstractions make sense for any repetitions on this scale. The additional effort is minimal and any tech debt is lifted while it's low. We can already surmise from the designs that JSON viewers etc are reused in other places.

Reply:
Yep, it looks good. Only one comment in your PR.